### PR TITLE
build: update dependency jasmine to v5

### PR DESCRIPTION
**Pull request by alan-agius4**
The patch is to fix upstream bug in `@bazel/jasmine`. See: bazelbuild/rules_nodejs#3691

### CODE REVIEW

File: tools/esm-interop/patches/npm/@bazel+jasmine+5.8.1.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
++++ b/node_modules/@bazel/jasmine/jasmine_runner.js
+@@ -147,7 +147,7 @@ async function main(args) {
+   // TODO(6.0): remove support for deprecated versions of Jasmine that use the old API &
+   // remember to update the `peerDependencies` as well.
+   // Jasmine versions prior to 4.0.0 should use the old API.
+-  if (jrunner.coreVersion().charAt(0) !== '4') {
++  if (+jrunner.coreVersion().charAt(0) < 4) {
+     console.warn(`DEPRECATED: Support for Jasmine versions prior to '4.0.x' is deprecated in '@bazel/jasmine'.`);
+ 
+     // Old Jasmine API.
```
Comment (devversion) [inline tools/esm-interop/patches/npm/@bazel+jasmine+5.8.1.patch:13]: Do you think we can have a single patch file and use it for both AIO + main? also feels a little weird having that file in the esm interop

File: tools/testing/BUILD.bazel [hunk 2]
```diff
@@ -71,5 +73,7 @@ jasmine_node_test(
     name = "fail_bootstrap_test",
     srcs = ["fail.spec.js"],
     bootstrap = ["//tools/testing:node"],
-    expected_exit_code = 55,
+    # While the force the termination of the process with an exitCode of 55 in fail.spec.js. Jasmine force it to 4.
```
Comment (devversion) [inline tools/testing/BUILD.bazel:76]:
```suggestion
    # While we force the termination of the process with an exitCode of 55 in fail.spec.js. Jasmine force it to 4.
```

**Review by devversion — APPROVED** [review]
LGTM overall, just two minor comments

**Review by josephperrott — APPROVED** [review]
LGTM

File: tools/esm-interop/patches/npm/@bazel+jasmine+5.8.1.patch (hunk 1 above)
Comment (alan-agius4) [reply tools/esm-interop/patches/npm/@bazel+jasmine+5.8.1.patch]: Moving into a new directory and shared it with AIO.

File: aio/package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "author": "Angular",
   "license": "MIT",
   "scripts": {
-    "postinstall": "node tools/cli-patches/patch.js",
+    "postinstall": "node tools/cli-patches/patch.js && patch-package --patch-dir ../tools/npm-patches/",
```
Comment (devversion) [inline aio/package.json:9]: OOC: do we need to wire up the same in the top-level `package.json`? (`tools/npm-patches`). Is it green right now by accident? (due to the caching issue we've seen with patches after CircleCI migration?)

Comment (alan-agius4) [reply aio/package.json:9]: Good call

Comment (devversion) [reply aio/package.json:9]: did it just work due to caching? does this show something is wrong?

Comment (alan-agius4) [reply aio/package.json:9]: Actually it was not green.

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 0ee0f780e4c19d43dde9bac0bb6468ea6431d24b.

**Pull request closed without merging**